project/src/main/java/com/google/sps/servlets/BlobHandlerEventServlet.java

```diff
@@ -0,0 +1,80 @@
+import com.google.sps.data.Event;
+import com.google.sps.utilities.SpannerTasks;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * When the user submits the form, Blobstore processes the file upload and then forwards the request
+ * to this servlet. This servlet can then process the request using the file URL we get from
+ * Blobstore.
+ */
+@WebServlet("/event-blob-handler")
+public class BlobHandlerEventServlet extends HttpServlet {
+  private static final String EVENT_PICTURE = "event-picture";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Optional<Event> eventOptional = SpannerTasks.getEventById(request.getParameter("event-id"));
+    if (eventOptional.isPresent()) {
+      Event event = eventOptional.get();
+      response.getWriter().println(event.getImageUrl());
+    } else {
+      response.sendError(
+          HttpServletResponse.SC_BAD_REQUEST, "Error with getting event: does not exist");
+    }
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String eventId = request.getParameter("event-id");
+    Optional<Event> eventOptional = SpannerTasks.getEventById(eventId);
+    if (eventOptional.isPresent()) {
+      Event event = eventOptional.get();
+      String blobKeyString = getUploadedBlobKeyString(request);
+      SpannerTasks.insertorUpdateEvent(event.toBuilder().setImageUrl(blobKeyString).build());
+    } else {
+      response.sendError(
+          HttpServletResponse.SC_BAD_REQUEST, "Error with getting event: does not exist");
+    }
+    response.sendRedirect(String.format("/event-details.html?eventId=%s", eventId));
+  }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+  private static String getUploadedBlobKeyString(HttpServletRequest request) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(EVENT_PICTURE);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+    BlobKey blobKey = blobKeys.get(0);
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      return null;
+    }
+    return blobKey.getKeyString();
```

Comment:
I think v similar or exact code for profile blob image? maybe common utils

Reply:
Done